doc/api/vm.md

```diff
@@ -13,66 +14,86 @@ const vm = require('vm');
 
-## Class: Script
+***Note*: Great care should be taken when running potentially untrusted code
+within the same process as trusted or privileged code. While the `vm` module
+provides valuable functionality, safely running untrusted code requires a
+separate process.**
 
-A class for holding precompiled scripts, and running them in specific sandboxes.
+## Class: vm.Script
+
+Instances of the `vm.Script` class contained precompiled scripts that can be
+executed in specific sandboxes (or "contexts").
 
 ### new vm.Script(code, options)
 
-Creating a new `Script` compiles `code` but does not run it. Instead, the
-created `vm.Script` object represents this compiled code. This script can be run
-later many times using methods below. The returned script is not bound to any
-global object. It is bound before each run, just for that run.
-
-The options when creating a script are:
-
-- `filename`: allows you to control the filename that shows up in any stack
-  traces produced from this script.
-- `lineOffset`: allows you to add an offset to the line number that is
-  displayed in stack traces
-- `columnOffset`: allows you to add an offset to the column number that is
-  displayed in stack traces
-- `displayErrors`: if `true`, on error, attach the line of code that caused
-  the error to the stack trace. Applies only to syntax errors compiling the
-  code; errors while running the code are controlled by the options to the
-  script's methods.
-- `timeout`: a number of milliseconds to execute `code` before terminating
-  execution. If execution is terminated, an [`Error`][] will be thrown.
-- `cachedData`: an optional `Buffer` with V8's code cache data for the supplied
-  source. When supplied `cachedDataRejected` value will be set to either
-  `true` or `false` depending on acceptance of the data by V8.
-- `produceCachedData`: if `true` and no `cachedData` is present - V8 tries to
-  produce code cache data for `code`. Upon success, a `Buffer` with V8's code
-  cache data will be produced and stored in `cachedData` property of the
-  returned `vm.Script` instance. `cachedDataProduced` value will be set to
-  either `true` or `false` depending on whether code cache data is produced
-  successfully.
+* `code` {string} The JavaScript code to compile.
+* `options`
+  * `filename` {string} Specifies the filename used in any stack traces
+    produced from this script.
+  * `lineOffset` {number} Specifies the line number offset that is displayed
+    in stack traces produced from this script.
+  * `columnOffset` {number} Specifies the column number offset that is displayed
+    in stack traces produced from this script.
+  * `displayErrors` {boolean} When `true`, when a `SyntaxError` error occurs
+    while compiling the `code`, the line of code causing the error is attached
+    to the stack trace.
+  * `timeout` {number} Specifies the number of milliseconds to execute `code`
+    before terminating execution. If execution is terminated, an [`Error`][]
```

Comment:
If the code completes before the timeout, all is well. If the code does not complete before the timer fires, an Error is thrown